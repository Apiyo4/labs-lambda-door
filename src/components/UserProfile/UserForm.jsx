import React from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';

const Input = styled.input`
  margin-bottom: 1%;
  width: 100%;
  padding: 12px 20px;
  display: inline-block;
  border: 1px solid #ccc;
  border-radius: 4px;
  box-sizing: border-box;
`;
const H4 = styled.h4`
  font-weight: bold;
`;

const Button = styled.button`
border: 2px solid #7C9E9A;
padding:12px 20px;
    `
export const UserForm = ({ credentials }) => {
  return (
    <div>
      <form>
        <H4>Full Name:</H4>
        {/* <Input value={credentials.full_name} /> */}
        <br />
        <H4>Username:</H4>
        <Input placeholder="josiah-williams" />
        <br />
        <H4>Email:</H4>
        <Input placeholder="josiahdamiwilliams@gmail.com" />
        <br />
        <H4>Age:</H4>
        <Input placeholder="20" />
        <br />
        <H4>Location:</H4>
        <Input placeholder="Nigeria" />
        <br />
        <H4>Github Link:</H4>
        <Input />
        <br/>
        <H4>Linkedin Link:</H4>
        <Input/>
        <br/>
        <H4>Portfolio Link:</H4>
        <Input/>
        <br/>
        <Button>Update Profile</Button>
      </form>
    </div>
  );
};


// const mapStateToProps = state => ({
//   credentials: state.authState.credentials,
//   isLoading: state.authState.isLoading
// });

// export default connect(mapStateToProps)(UserForm);