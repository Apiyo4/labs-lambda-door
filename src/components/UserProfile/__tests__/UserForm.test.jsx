import React from 'react';
import * as rtl from '@testing-library/react';
import UserForm from '../../UserProfile/UserForm';
import { renderWithRedux } from '../../../utils/testingHelpers';

beforeEach(rtl.cleanup);

describe('UserForm', () =>{
    it('renders without crashing', ()=>{
        renderWithRedux(<UserForm/>);
    });

    it ('renders correctly', ()=>{
        expect(renderWithRedux(<UserForm/>).baseElement).toMatchSnapshot();
    });

    it('asserts that the component renders properly',()=>{
        expect(renderWithRedux(<UserForm/>).baseElement).toMatchSnapshot();
    })

});