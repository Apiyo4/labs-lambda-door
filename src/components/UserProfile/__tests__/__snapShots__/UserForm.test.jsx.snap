// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm asserts that the component renders properly 1`] = `
<body>
  <div>
    <div>
      <form>
        <h4
          class="sc-bwzfXH hZEuEv"
        >
          Full Name:
        </h4>
        <input
          class="sc-bdVaJa kVTPwc"
          name="full_name"
          type="text"
          value=""
        />
        <br />
        <h4
          class="sc-bwzfXH hZEuEv"
        >
          Username:
        </h4>
        <input
          class="sc-bdVaJa kVTPwc"
          name="username"
          type="text"
          value=""
        />
        <br />
        <h4
          class="sc-bwzfXH hZEuEv"
        >
          Email:
        </h4>
        <input
          class="sc-bdVaJa kVTPwc"
          name="email_address"
          type="email"
          value=""
        />
        <br />
        <h4
          class="sc-bwzfXH hZEuEv"
        >
          Age:
        </h4>
        <input
          class="sc-bdVaJa kVTPwc"
          name="age"
          type="number"
          value=""
        />
        <br />
        <h4
          class="sc-bwzfXH hZEuEv"
        >
          Location:
        </h4>
        <input
          class="sc-bdVaJa kVTPwc"
          name="location"
          type="text"
          value=""
        />
        <br />
        <h4
          class="sc-bwzfXH hZEuEv"
        >
          Github Link:
        </h4>
        <input
          class="sc-bdVaJa kVTPwc"
          name="github_link"
          type="text"
          value=""
        />
        <br />
        <h4
          class="sc-bwzfXH hZEuEv"
        >
          Linkedin Link:
        </h4>
        <input
          class="sc-bdVaJa kVTPwc"
          name="linkedin_link"
          type="text"
          value=""
        />
        <br />
        <h4
          class="sc-bwzfXH hZEuEv"
        >
          Portfolio Link:
        </h4>
        <input
          class="sc-bdVaJa kVTPwc"
          name="portfolio_link"
          type="text"
          value=""
        />
        <br />
        <button
          class="sc-htpNat llMOo"
        >
          Update Profile
        </button>
      </form>
    </div>
  </div>
</body>
`;

exports[`UserForm renders correctly 1`] = `
<body>
  <div>
    <div>
      <form>
        <h4
          class="sc-bwzfXH hZEuEv"
        >
          Full Name:
        </h4>
        <input
          class="sc-bdVaJa kVTPwc"
          name="full_name"
          type="text"
          value=""
        />
        <br />
        <h4
          class="sc-bwzfXH hZEuEv"
        >
          Username:
        </h4>
        <input
          class="sc-bdVaJa kVTPwc"
          name="username"
          type="text"
          value=""
        />
        <br />
        <h4
          class="sc-bwzfXH hZEuEv"
        >
          Email:
        </h4>
        <input
          class="sc-bdVaJa kVTPwc"
          name="email_address"
          type="email"
          value=""
        />
        <br />
        <h4
          class="sc-bwzfXH hZEuEv"
        >
          Age:
        </h4>
        <input
          class="sc-bdVaJa kVTPwc"
          name="age"
          type="number"
          value=""
        />
        <br />
        <h4
          class="sc-bwzfXH hZEuEv"
        >
          Location:
        </h4>
        <input
          class="sc-bdVaJa kVTPwc"
          name="location"
          type="text"
          value=""
        />
        <br />
        <h4
          class="sc-bwzfXH hZEuEv"
        >
          Github Link:
        </h4>
        <input
          class="sc-bdVaJa kVTPwc"
          name="github_link"
          type="text"
          value=""
        />
        <br />
        <h4
          class="sc-bwzfXH hZEuEv"
        >
          Linkedin Link:
        </h4>
        <input
          class="sc-bdVaJa kVTPwc"
          name="linkedin_link"
          type="text"
          value=""
        />
        <br />
        <h4
          class="sc-bwzfXH hZEuEv"
        >
          Portfolio Link:
        </h4>
        <input
          class="sc-bdVaJa kVTPwc"
          name="portfolio_link"
          type="text"
          value=""
        />
        <br />
        <button
          class="sc-htpNat llMOo"
        >
          Update Profile
        </button>
      </form>
    </div>
  </div>
</body>
`;
